pkg/kubelet/cm/device_plugin_handler_test.go

```diff
@@ -128,6 +128,42 @@ func (m *DevicePluginManagerTestStub) Stop() error {
 	return nil
 }
 
+func (m *DevicePluginManagerTestStub) CheckpointFile() string {
+	fmt.Printf("call CheckpointFile")
+	return "/tmp/device-plugin-checkpoint"
+}
+
+func TestCheckpoint(t *testing.T) {
+	resourceName1 := "domain1.com/resource1"
+	resourceName2 := "domain2.com/resource2"
+
+	m, err := NewDevicePluginManagerTestStub()
+	as := assert.New(t)
+	as.Nil(err)
+
+	testDevicePluginHandler := &DevicePluginHandlerImpl{
+		devicePluginManager: m,
+		allDevices:          make(map[string]sets.String),
+		allocatedDevices:    make(map[string]podDevices),
+	}
+	testDevicePluginHandler.allocatedDevices[resourceName1] = make(podDevices)
+	testDevicePluginHandler.allocatedDevices[resourceName1].insert("pod1", "con1", "dev1")
+	testDevicePluginHandler.allocatedDevices[resourceName1].insert("pod1", "con1", "dev2")
+	testDevicePluginHandler.allocatedDevices[resourceName1].insert("pod1", "con2", "dev1")
```

Comment:
Not that it matters for this test but I found using `dev1` again in this line and the next line a bit confusing at first.